src/animation/use-animated-state.ts

```diff
@@ -1,67 +1,56 @@
-import { useState, useEffect } from "react"
-import { useValueAnimationControls } from "./use-value-animation-controls"
-import { useMotionValues } from "../motion/utils/use-motion-values"
-import { AnimationDefinition } from "./ValueAnimationControls"
+import { useEffect, useState } from "react"
+import { VisualElement } from "../render/VisualElement"
 import { useConstant } from "../utils/use-constant"
+import { useVisualElementAnimation } from "./use-visual-element-animation"
+import { AnimationDefinition } from "./VisualElementAnimationControls"
+import { ResolvedValues } from "../render/types"
+
+class StateVisualElement extends VisualElement {
+    latestState: ResolvedValues = {}
+
+    build() {}
+    clean() {}
+
+    getBoundingBox() {
+        return { x: { min: 0, max: 0 }, y: { min: 0, max: 0 } }
+    }
+
+    readNativeValue(key: string) {
+        return this.latestState[key] || 0
+    }
+
+    render() {
+        console.log("render", this.latest.foo)
+    }
+}
```

Comment:
Is this WIP code? Can it be committed safely?

Reply:
Ah yeah good catch on the render. This is basically just some hack to reimplement useAnimatedState using the new internal APIs. This isn't a public API, but some Framer users were told about it and now use it in their projects.

Reply:
Cool 👍